File: app/src/main/java/com/u1fukui/bbs/ui/creation/thread/CreateThreadNavigator.java
package com.u1fukui.bbs.ui.creation.thread;


import com.u1fukui.bbs.model.Category;
import com.u1fukui.bbs.ui.Navigator;

import javax.inject.Inject;

public class CreateThreadNavigator extends Navigator {

    @Inject
    public CreateThreadNavigator(CreateThreadActivity activity) {
        super(activity);
    }

    public void navigateToInputInfoPage(Category category) {
        if (getActivity() instanceof CreateThreadActivity) {
            CreateThreadActivity activity = (CreateThreadActivity) getActivity();
            activity.getSupportFragmentManager().beginTransaction()
                    .replace(activity.getFragmentContainerId(), InputThreadInfoFragment.newInstance(category))
                    .addToBackStack(null)
                    .commit();
        }
    }
}

File: app/src/main/java/com/u1fukui/bbs/ui/creation/thread/CreateThreadNavigator.kt
package com.u1fukui.bbs.ui.creation.thread


import com.u1fukui.bbs.model.Category
import com.u1fukui.bbs.ui.Navigator

import javax.inject.Inject

class CreateThreadNavigator @Inject
constructor(activity: CreateThreadActivity) : Navigator(activity) {

    fun navigateToInputInfoPage(category: Category) {
        if (activity is CreateThreadActivity) {
            val activity = activity
            activity.supportFragmentManager.beginTransaction()
                    .replace(activity.fragmentContainerId, InputThreadInfoFragment.newInstance(category))
                    .addToBackStack(null)
                    .commit()
        }
    }
}
